iam/terraform/account-specific/main/elasticsearch.tf

```diff
@@ -143,6 +143,17 @@ resource "aws_cognito_identity_pool_roles_attachment" "log_viewers" {
   }
 }
 
+resource "opensearch_snapshot_repository" "archived-logs" {
+  name = "archived-logs"
+  type = "s3"
+  settings = {
+    bucket   = "${var.log_snapshot_bucket_name}"
+    region   = "us-east-1"
+    role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/es-s3-snapshot-access"
+    # role_arn = aws_iam_role.es_s3_snapshot_access_role.arn
```

Comment:
is this still helpful? is this the one that should be using? looks incomplete

Reply:
Yeah I think this section needs to be fixed (as well as a section in `s3.tf`) but I'm not sure how.  

The commented variable `aws_iam_role.es_s3_snapshot_access_role.arn` resolves to `arn:aws:iam::{id goes here}:role/es_s3_snapshot_access_role`   Which is a resource that's defined in `s3.tf`.   

The not-commented line refers to `es-s3-snapshot-access`, which is referenced in `s3.tf` but not actually defined.    I assume we either need to choose one of these role names and then replace the references to the other, but I'm not sure which role name is "right".

Reply:
Okay I think this section is truly fixed this time in the latest commit. There were two issues:
- we were referring to the wrong ARN name (should be `es_s3_snapshot_access_role` not `es-s3-snapshot-access`
- we used string interpolation to specify the `role_arn` value instead of using terraforms expression syntax (`aws_iam_role.es_s3_snapshot_access_role.arn`).   Originally me and Jim thought this line was causing an error, but upon a further look it turns out not to be the case.